trieste/acquisition/function.py

```diff
@@ -1385,6 +1265,138 @@ def update_acquisition_function(
         )
 
 
+class BatchMonteCarloExpectedHypervolumeImprovement(SingleModelGreedyAcquisitionBuilder):
```

Comment:
Would it be possible to move this class back where it used to be within this file? This diff makes it difficult to see what changes were actually made to the implementation of this acquisition.

Reply:
Indeed, though it seems not very easy to move back as we have changed the super class (i.e., now it shall be defined after `SingleModelGreedyAcquisitionBuilder`), please let me know if there is some elegant way to show the changes more clearly.

i may highlight the changes made in this PR, which is actually restricted to only several lines of code:
- add `pending_points` at [1303](https://github.com/TsingQAQ/trieste/blob/e532d7212a5adfd894d48d5610f0b54cf367fc66/trieste/acquisition/function.py#L1303)/[1330](https://github.com/TsingQAQ/trieste/blob/e532d7212a5adfd894d48d5610f0b54cf367fc66/trieste/acquisition/function.py#L1330) to store those chosen yet evaluated location
- main change is: *switch [1061](https://github.com/secondmind-labs/trieste/blob/a4b86cf58781ffb2e97760f9f42bc737ee987a68/trieste/acquisition/function.py#L1061) to [1353-1359](https://github.com/TsingQAQ/trieste/blob/e532d7212a5adfd894d48d5610f0b54cf367fc66/trieste/acquisition/function.py#L1353-L1359)*: instead of doing [1359](https://github.com/TsingQAQ/trieste/blob/e532d7212a5adfd894d48d5610f0b54cf367fc66/trieste/acquisition/function.py#L1359) as before, if we have pending points, we now concat them with current candidate `at` and sample them together to get the joint distribution 

If this is still not easy to review, maybe i can temporarily move `batch_ehvi` back, which is able to = show the difference partially, please let me know which suits better.

Reply:
Now it seems more easier to see the diff.